tasks/main.yml

```diff
@@ -0,0 +1,104 @@
+      when: ansible_facts.services["ovirt-engine.service"].state != 'running'
+
+    - name: Login to oVirt
+      ovirt_auth:
+        url: "{{ engine_url | default(lookup('env','OVIRT_URL')) | default(omit) }}"
+        username: "{{ engine_user | default(lookup('env','OVIRT_USERNAME')) | default(omit) }}"
+        password: "{{ engine_password | default(lookup('env','OVIRT_PASSWORD')) | default(omit) }}"
+        ca_file: "{{ engine_cafile | default(lookup('env','OVIRT_CAFILE')) | default(omit) }}"
+        token: "{{ engine_token | default(lookup('env','OVIRT_TOKEN')) | default(omit) }}"
+        insecure: "{{ engine_insecure | default(true) }}"
+      when: ovirt_auth is undefined or not ovirt_auth
+      register: loggedin
+      tags:
+        - always
+
+    - name: Get hosts
+      ovirt_hosts_facts:
+        auth: "{{ ovirt_auth }}"
+        all_content: true
+
+    - set_fact:
+        he_hosts: "{{ ovirt_hosts | selectattr('hosted_engine', 'defined') | selectattr('hosted_engine.configured') | list }}"
+
+    - set_fact:
+        non_he_hosts: "{{ ovirt_hosts | difference(he_hosts) }}"
+
+    - set_fact:
+        non_he_hosts_ipmi: "{{ non_he_hosts | selectattr('power_management', 'defined') | selectattr('power_management.enabled') | list }}"
+
+    - set_fact:
+        non_he_hosts_noipmi: "{{ non_he_hosts | difference(non_he_hosts_ipmi) }}"
+
+    - name: Get VM list
+      ovirt_vms_facts:
+        auth: "{{ ovirt_auth }}"
+        all_content: true
+
+    - name: Shutdown all VMs, except HostedEngine
+      ovirt_vms:
+        state: stopped
+        name: "{{ item.name }}"
+        auth: "{{ ovirt_auth }}"
+      when: "item.origin != 'managed_hosted_engine'"
+      with_items:
+        - "{{ ovirt_vms }}"
+
+    - name: Shutdown hosts, except HE ones, via IPMI (if configured)
+      ovirt_hosts:
+        state: stopped
+        name: "{{ item.name }}"
+        auth: "{{ ovirt_auth }}"
+      with_items:
+        - "{{ non_he_hosts_ipmi }}"
+
+    - name: Shutdown hosts, except HE ones, via IPMI (if configured)
+      command: ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i /etc/pki/ovirt-engine/keys/engine_id_rsa -p {{item.ssh.port}} -t root@{{item.address}} 'while pgrep qemu-kvm >/dev/null; do sleep 5; done; shutdown -h now'
+      async: 1000
+      poll: 0
+      with_items:
+        - "{{ non_he_hosts_noipmi }}"
+
+    - name: Set global maintenance mode
+      command: ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i /etc/pki/ovirt-engine/keys/engine_id_rsa -p {{item.ssh.port}} -t root@{{item.address}} 'hosted-engine --set-maintenance --mode=global'
+      with_items:
+        - "{{ he_hosts }}"
+
+    - name: Shutdown of HE hosts
+      command: ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i /etc/pki/ovirt-engine/keys/engine_id_rsa -p {{item.ssh.port}} -t root@{{item.address}} 'while pgrep qemu-kvm >/dev/null; do sleep 5; done; shutdown -h now'
```

Comment:
This is not enough in HC case since the engine VM is running just on one of the three hosts